### DISCUSSION TITLE
Frosted Glass / blur mask effect?

### DISCUSSION THREAD

**Discussion by reececomo** [pixijs/pixijs#10201] [Q&A] [answered]
Would it be challenging to create rectangle that blurs everything behind it?

...that updates at least 15-30 times per second. Mobile support optional. Ideally less than 1-2 draws too.

#### Desired before → after

![before](https://github.com/pixijs/pixijs/assets/7390781/209b0b86-7b56-47a0-9f6c-a22ed9b7eaff) ![after](https://github.com/pixijs/pixijs/assets/7390781/d0afd139-1459-4a8e-8d66-c55076737163)

#### Other examples
* CSS + PIXI: https://codepen.io/georgedoescode/pen/XWNmvro
* CSS (non-PIXI): https://webdesign.tutsplus.com/how-to-create-a-frosted-glass-effect-in-css--cms-32535t

**Comment by gabrielliwerant** [comment]
One way you could go about this is to create a graphic or sprite and then use it as a mask with a blur filter, so it should only blur the parts of the background it hovers over: https://pixijs.com/examples/masks/filter. I don't think it would be too challenging, depending on how complex your use case is.

**Reply by reececomo** [reply]
Hey @gabrielliwerant, I appreciate your fast response! I'm not sure that what you've described produces the effect we're after.

![filter-blur](https://github.com/pixijs/pixijs/assets/7390781/99dd9215-c7f6-466c-bca1-459e7d4f4c71)

I'm not trying to blur the mask itself, but rather everything underneath it.

**Reply by reececomo** [reply]
Although that will work great for a vignette effect, which I'm sure will be useful in other areas! Appreciate it!

**Reply by gabrielliwerant** [reply]
The link is just an example of what can be accomplished with masks and filters. You can achieve what you want by adapting something like that to a blur effect (and visible background). It is just to quickly demonstrate that the functionality exists in the core features of pixi. It looks like you did eventually find what you were after, so glad you were able to figure it out.

**Reply by reececomo** [reply]
I'm with you. I imagine this specific case was a "solved problem", so we were after something more direct. Thanks in either case.

**Comment by reececomo** [comment, marked as answer]
Looks like [@pixi/picture](https://github.com/pixijs/picture)'s MaskFilter might have what we're after --> https://api.pixijs.io/@pixi/picture/MaskFilter.html

**Update:** Success ✅ 

### Code

Here's some sample code for the next person:

```ts
import { BlurFilter, Container, Graphics } from 'pixi.js';
import { MaskFilter } from '@pixi/picture';

export class FrostedGlassPill extends Container {
  public constructor(width: number, height: number) {
    super();

    // Create the blur mask shape
    const mask = new Graphics()
      .beginFill(0xffffff)
      .drawRoundedRect(0, 0, width, height, Math.min(width, height) / 2);

    // Create a white overlay
    const overlay = new Graphics()
      .beginFill(0xffffff)
      .drawRoundedRect(0, 0, width, height, Math.min(width, height) / 2);
    overlay.alpha = 0.5;

    // Apply mask / blur to the mask shape.
    const filter = new MaskFilter(new BlurFilter());
    filter.safeFlipY = false; // performance, see @pixi/picture docs
    mask.filters = [filter];

    this.addChild(mask, overlay);
  }
}
```

(And don't forget to set `backgroundAlpha: 0` in your app.renderer config, like I did 😆)

### Output:

![result](https://github.com/pixijs/pixijs/assets/7390781/216833c9-1803-4924-9241-0cc81587854c)
